File: cmd/unix_socket_client/unix_socket_client.js
const net = require('net');

// 定义对象结构
class Message {
    constructor(text, id) {
        this.text = text;
        this.id = id;
    }
}

const socketPath = '/tmp/unix_socket';
const client = net.createConnection(socketPath);

client.on('connect', () => {
    // 发送对象
    const msg = new Message('Hello, server!', 1);
    client.write(JSON.stringify(msg));
});

client.on('data', (data) => {
    const resp = JSON.parse(data);
    console.log('Received response:', resp);
    client.end();
});

client.on('error', (err) => {
    console.error('Connection error:', err);
});

client.on('end', () => {
    console.log('Disconnected from server');
});

File: cmd/unix_socket_client/unix_socket_client.py
import json
import socket
import os

# 定义对象结构
class Message:
    def __init__(self, text, msg_id):
        self.text = text
        self.id = msg_id

def main():
    socket_path = "/tmp/unix_socket"

    # 创建Unix Socket
    client = socket.socket(socket.AF_UNIX, socket.SOCK_STREAM)
    client.connect(socket_path)

    # 发送对象
    msg = Message("Hello, server!", 1)
    msg_json = json.dumps(msg.__dict__)
    client.sendall(msg_json.encode('utf-8'))

    # 接收回应
    data = client.recv(1024)
    resp_json = data.decode('utf-8')
    resp = json.loads(resp_json)

    print(f"Received response: {resp}")

    client.close()

if __name__ == "__main__":
    main()
